frontend/views/containers/contributions/NonMonetaryPledges.vue

```diff
@@ -0,0 +1,174 @@
+<template lang='pug'>
+fieldset(
+  data-test='nonMonetaryPledges'
+  v-error:pledges='{ attrs: { "data-test": "badPledges" } }'
+)
+  legend.has-text-bold.c-legend
+    i18n.is-title-4 Non-monetary pledge
+    i18n.c-optional(v-if='optional') (optional)
+
+  i18n.has-text-1 All members can support each other with non-monetary contributions. There's value in time, skills, and willingness to help the group.
+
+  ul.c-fields(data-test='nonMonetaryFields')
+    li.c-fields-item(
+      v-for='(pledge, index) in form.pledges'
+      :key='`pledge-${pledge.id}`'
+      data-test='pledgeEntry'
+    )
+      fieldset
+        .inputgroup
+          input.input(
+            type='text'
+            v-model='pledge.value'
+            data-test='inputNonMonetaryPledge'
+            :maxlength='config.maxChar'
+            :aria-label='L("Pledge value")'
+            :class='{ error: $v.form.pledges.$each[index].value.$error }'
+          )
+          button.is-icon-small.is-btn-shifted(
+            type='button'
+            :aria-label='L("Remove pledge entry")'
+            @click='removeEntry(index)'
+            data-test='removePledgeEntry'
+          )
+            i.icon-times
+
+        span.error(v-if='$v.form.pledges.$each[index].value.$error') {{ getFieldErrorMsg(index) }}
+
+  button.link.has-icon(
+    type='button'
+    @click='addPledgeEntry'
+    data-test='addPledgeEntry'
+  )
+    i.icon-plus
+    i18n Add more
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import { L } from '@common/common.js'
+import { validationMixin } from 'vuelidate'
+import { maxLength } from 'vuelidate/lib/validators'
+import { GROUP_NON_MONETARY_CONTRIBUTION_MAX_CHAR } from '@model/contracts/shared/constants.js'
+import { randomHexString } from '@model/contracts/shared/giLodash.js'
+
+export default {
+  name: 'NonMonetaryPledges',
+  mixins: [validationMixin],
+  props: {
+    optional: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      form: {
+        pledges: [
+          { id: randomHexString(10), value: '' }
```

Comment:
Is this `id` thing necessary? If all it's used for is is the `:key` in the `v-for`, then can't the position (`index`) in the array be used instead?

Reply:
@taoeffect Can we keep this as is? We have a feature to add/remove an array item and in this case, having unique id like this can help prevent duplicated keys from being generated accidentally.

Reply:
yeah, I was was just trying to understand why it was necessary